_specs/ecip-proto-treasury-system.md

```diff
@@ -0,0 +1,450 @@
+    /// @param participant address from which we desire to query the amount of funds
+    /// @return amount locked with the participant
+    function getLockedAmount(address participant) external view returns(uint256);
+
+    /// @notice Returns the amount of active members of client type.
+    function getClientMembersSize() external view returns(uint256);
+    
+    /// @notice Returns the address and the name of the Client member in the index array position
+    function getClientMemberAt(uint256 index) external view returns(address, string memory);
+    
+    /// @notice Returns the address of the gitcoin members.
+    function getGitcoinAddress() external view returns(address);
+
+    /// @notice Returns the state of a proposal
+    /// @return active whether the proposal was executed/closed or not
+    /// @return preSupport whether the proposal has been pre-approved or not
+    /// @return proposedAtTime time when the proposal was created
+    /// @return endsAtTime time when the proposal voting period will end and it will be open to closing/executing
+    /// @return yesVotes amount of stake in favour of the proposal
+    /// @return noVotes amount of stake against a proposal
+    function getProposalState(uint256 proposalId) external view returns(bool active, bool preSupport, uint256 proposedAtTime, uint256 endsAtTime, uint256 yesVotes, uint256 noVotes);
+
+    /// @notice true if there a proposal for shutdown was executed 
+    function withShutdownInProgress() external view returns(bool);
+}
+```
+
+#### Notes and considerations
+
+##### Client account maintenance
+
+In the event that a client funding account has been lost or compromised, we had incorporated a proposal type (*ProposalType.UpdateClientAddress*) to make the change, but going through the proposal process to make such a update might be overkill and there is no need to involve the whole community in something that is solely the client's responsibility.
+
+We suggest an alternative mechanism to cover this, in which each client would have two different sets of keys; one **spending key** that would allow them to manage their funds and acts as a *hot wallet* and a second one that can act as a **revocation key/certificate**. This second key doesn't neccessarily have to be an ETC account but could follow another standard (such as PGP for example).
+This would facilitate the client in making a more frequent rotation of their spending keys in a secure way.
+
+##### Ambush attacks prevention
+
+**Ambush attack**: "Surprise" that a huge approval is given to a proposal at the last minute. A proposal that seems to be going towards a "No" outcome, suddenly receives a big amount of yes votes and the result is flipped at the last minute.
+
+**Prevention**: Successful proposals require early pre-support, which already raises warnings that this approval might go through, so anyone that wants to counteract, can intervene during this period.
```

Comment:
> _require early pre-support_

is this specified somehow?

Reply:
I clarified the pre-support event section for the proposal life-cycle. it's also included in the _Proposal approval criteria_ list, as well as implemented in the reference attached implementation. Let me know if you feel something in particular is not clear enough around this. ;)